constraint-solver/src/solver/exhaustive_search.rs

```diff
@@ -0,0 +1,152 @@
+use itertools::Itertools;
+use powdr_number::{FieldElement, LargeInt};
+
+use crate::quadratic_symbolic_expression::RangeConstraintProvider;
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::{Debug, Display};
+use std::hash::Hash;
+
+use super::{Error, Solver};
+
+/// The maximum number of possible assignments to try when doing exhaustive search.
+const MAX_SEARCH_WIDTH: u64 = 1 << 12;
+
+pub struct ExhaustiveSearch<'a, T: FieldElement, V> {
+    solver: &'a Solver<T, V>,
+}
+
+impl<'a, T: FieldElement, V> ExhaustiveSearch<'a, T, V> {
+    pub fn new(solver: &'a Solver<T, V>) -> Self {
+        ExhaustiveSearch { solver }
+    }
+}
+
+impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> ExhaustiveSearch<'_, T, V> {
+    /// Tries to find unique assignments via exhaustive search: For any group of variables that
+    /// appear together in an identity, if there are fewer than `MAX_SEARCH_WIDTH` possible
+    /// assignments, it tries them all and returns any unique assignments.
```

Comment:
```suggestion
    /// assignments, it tries them all and returns any unique non-conflicting assignments.
```

Reply:
Hmm, that's clear from the type no? A map can't have the same key twice. There's also already a sentence below that describes the error case.